### ISSUE TITLE
export splitter as a utility

### ISSUE THREAD

**Issue by nikomatsakis** [nikomatsakis/rayon#112]
We should have some easy way for "divide-and-conquer" algorithms like quicksort to take advantage of the new `ThiefSplitter` stuff added by @cuviper in #106. I envision a wrapper around `join` that is like `splitter.join(x, y)` which will run `x` and `y` sequentially after a certain point.

**Comment by nikomatsakis** [comment]
To validate the design, we should port the quicksort and mergesort benchmarks to use it.

**Comment by nikomatsakis** [comment]
Also relevant: things like `chain` which call `drive` twice, but not in parallel with one another.

**Comment by cuviper** [comment]
Yeah.  I tried once to make `chain` use an unconditional `join` for that, but `ParallelIterator` doesn't have `Send`.  I think we'd have to go down to `Producer`s before we can split in parallel.

**Comment by Eh2406** [comment]
Random person on the internet looking forward to this functionality! 👍 :-)
